lib/BannerPlugin.js

```diff
@@ -22,21 +22,33 @@ const wrapComment = str => {
 
 class BannerPlugin {
 	constructor(options) {
-		if (arguments.length > 1)
+		if (arguments.length > 1) {
 			throw new Error(
 				"BannerPlugin only takes one argument (pass an options object)"
```

Comment:
Perhaps, the error message could be updated?

```js
throw new Error(
  "BannerPlugin only takes one argument (pass an options object or a function that returns an options object)"
);
```

Reply:
yep send a separate PR

Reply:
A function returning an options object is not supported

Reply:
Thanks; for some reason I was envisioning the constructor accepting the function. But it definitely makes more since that it's the banner property that accepts a function instead.